src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  static newAlive() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (!victim.hidden && victim instanceof Herbivore) {
+      victim.health -= 50;
+
+      if (victim.health <= 0) {
+        Animal.newAlive();
+      }
+    }
```

Comment:
The `bite` method correctly checks if the victim is not hidden and is an instance of `Herbivore`. However, it's better to filter the `Animal.alive` array inside this method after decreasing the health of the victim to ensure that the array is always up to date. The current implementation might lead to a delay in removing dead animals until the next `bite` call or manual invocation of `Animal.newAlive`.